internal/state/statuses.go

```diff
@@ -98,8 +102,9 @@ func buildStatuses(graph *graph) Statuses {
 		}
 
 		statuses.GatewayStatus = &GatewayStatus{
-			NsName:           client.ObjectKeyFromObject(graph.Gateway.Source),
-			ListenerStatuses: listenerStatuses,
+			NsName:             client.ObjectKeyFromObject(graph.Gateway.Source),
+			ListenerStatuses:   listenerStatuses,
+			ObservedGeneration: graph.Gateway.Source.Generation,
```

Comment:
should we also update https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3d94ea00eaab3ce692a055c0e0826a8013ab82fc/internal/state/statuses_test.go#L15 ?

Reply:
updated